<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Graded Assessment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a SQLIte database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. 

Use the links below to read the data files using the Pandas library. 

* Chicago Census Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Public Schools

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Crime Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

**NOTE:** Ensure you use the datasets available on the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**


Load the `pandas` and `sqlite3` libraries and establish a connection to `FinalDB.db`


In [1]:
!pip install pandas

Load the SQL magic module


In [14]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


Use `Pandas` to load the data available in the links above to dataframes. Use these dataframes to load data on to the database `FinalDB.db` as required tables.


Establish a connection between SQL magic module and the database `FinalDB.db`


In [15]:
%sql sqlite:///FinalDB

'Connected: @FinalDB'

In [39]:
import sqlite3
import pandas as pd
import csv
conn = sqlite3.connect('FinalDB.db')
cur = conn.cursor()

You can now proceed to the the following questions. Please note that a graded assignment will follow this lab and there will be a question on each of the problems stated below. It can be from the answer you received or the code you write for this problem. Therefore, please keep a note of both your codes as well as the response you generate.


## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [40]:

import sqlite3
import pandas as pd

# Load your data into Pandas DataFrames (replace these paths with your actual data)
census_data = pd.read_csv('./ChicagoCensusData.csv')
public_schools_data = pd.read_csv('./ChicagoPublicSchools.csv')
crime_data = pd.read_csv('./ChicagoCrimeData.csv')

# Establish a connection to the SQLite database (replace 'FinalDB.db' with your desired database name)
conn = sqlite3.connect('FinalDB.db')

# Write DataFrames to SQLite tables
census_data.to_sql('CENSUS_DATA', conn, index=False, if_exists='replace')
public_schools_data.to_sql('CHICAGO_PUBLIC_SCHOOLS', conn, index=False, if_exists='replace')
crime_data.to_sql('CHICAGO_CRIME_DATA', conn, index=False, if_exists='replace')

# Close the connection
conn.close()


In [37]:
type(data)

pandas.core.frame.DataFrame

In [42]:
%sql sqlite:///FinalDB



'Connected: @FinalDB'

In [46]:
import sqlite3
import pandas as pd

# Connect to the SQLite database
conn = sqlite3.connect('FinalDB.db')

# Query the first few rows of a table (replace TABLE_NAME with the actual table name)
query = "SELECT * FROM CENSUS_DATA LIMIT 5;"
result = pd.read_sql_query(query, conn)

# Display the result
print(result)

# Close the connection
conn.close()


   COMMUNITY_AREA_NUMBER COMMUNITY_AREA_NAME  PERCENT_OF_HOUSING_CROWDED  \
0                    1.0         Rogers Park                         7.7   
1                    2.0          West Ridge                         7.8   
2                    3.0              Uptown                         3.8   
3                    4.0      Lincoln Square                         3.4   
4                    5.0        North Center                         0.3   

   PERCENT_HOUSEHOLDS_BELOW_POVERTY  PERCENT_AGED_16__UNEMPLOYED  \
0                              23.6                          8.7   
1                              17.2                          8.8   
2                              24.0                          8.9   
3                              10.9                          8.2   
4                               7.5                          5.2   

   PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA  \
0                                          18.2   
1                               

In [78]:
%load_ext sql

# Connect to the SQLite database
%sql sqlite:///FinalDB.db

# SQL query to select the first few rows from the CENSUS_DATA table
result = %sql SELECT * FROM CENSUS_DATA LIMIT 5;

# Display the result
result


The sql extension is already loaded. To reload it, use:
  %reload_ext sql
   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[(1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0),
 (2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0),
 (3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0),
 (4.0, 'Lincoln Square', 3.4, 10.9, 8.2, 13.4, 25.5, 37524, 17.0),
 (5.0, 'North Center', 0.3, 7.5, 5.2, 4.5, 26.2, 57123, 6.0)]

In [48]:
import sqlite3

# Connect to the SQLite database
conn = sqlite3.connect('FinalDB.db')

# Specify the table name you want to read
table_name = 'CENSUS_DATA'  # Replace with the actual table name

# Query and fetch all rows from the table
query = f"SELECT * FROM {table_name};"
cursor = conn.cursor()
cursor.execute(query)

# Fetch all rows as a list of tuples
rows = cursor.fetchall()

# Display the result
for row in rows:
    print(row)

# Close the connection
conn.close()


(1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0)
(2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0)
(3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0)
(4.0, 'Lincoln Square', 3.4, 10.9, 8.2, 13.4, 25.5, 37524, 17.0)
(5.0, 'North Center', 0.3, 7.5, 5.2, 4.5, 26.2, 57123, 6.0)
(6.0, 'Lake View', 1.1, 11.4, 4.7, 2.6, 17.0, 60058, 5.0)
(7.0, 'Lincoln Park', 0.8, 12.3, 5.1, 3.6, 21.5, 71551, 2.0)
(8.0, 'Near North Side', 1.9, 12.9, 7.0, 2.5, 22.6, 88669, 1.0)
(9.0, 'Edison Park', 1.1, 3.3, 6.5, 7.4, 35.3, 40959, 8.0)
(10.0, 'Norwood Park', 2.0, 5.4, 9.0, 11.5, 39.5, 32875, 21.0)
(11.0, 'Jefferson Park', 2.7, 8.6, 12.4, 13.4, 35.5, 27751, 25.0)
(12.0, 'Forest Glen', 1.1, 7.5, 6.8, 4.9, 40.5, 44164, 11.0)
(13.0, 'North Park', 3.9, 13.2, 9.9, 14.4, 39.0, 26576, 33.0)
(14.0, 'Albany Park', 11.3, 19.2, 10.0, 32.9, 32.0, 21323, 53.0)
(15.0, 'Portage Park', 4.1, 11.6, 12.6, 19.3, 34.0, 24336, 35.0)
(16.0, 'Irving Park', 6.3, 13.1, 10.0, 22.4, 31.6, 27249, 34.0)
(17.0, 

In [49]:
import sqlite3

# Connect to the SQLite database
conn = sqlite3.connect('FinalDB.db')

# Specify the table name for which you want to count the rows
table_name = 'CHICAGO_CRIME_DATA'  # Replace with the actual table name

# Query to count the number of rows in the table
query = f"SELECT COUNT(*) FROM {table_name};"
cursor = conn.cursor()
cursor.execute(query)

# Fetch the result
count = cursor.fetchone()[0]

# Display the result
print(f"Number of rows in '{table_name}': {count}")

# Close the connection
conn.close()


Number of rows in 'CHICAGO_CRIME_DATA': 533


In [80]:
%load_ext sql

# Connect to the SQLite database
%sql sqlite:///FinalDB.db

# Specify the table name for which you want to count the rows
table_name = 'CHICAGO_CRIME_DATA'  # Replace with the actual table name

# Query to count the number of rows in the table
result = %sql SELECT COUNT(*) FROM $table_name;

# Display the result
count = result
print(f"Number of rows in '{table_name}': {count}")


The sql extension is already loaded. To reload it, use:
  %reload_ext sql
   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.
Number of rows in 'CHICAGO_CRIME_DATA': +----------+
| COUNT(*) |
+----------+
|   533    |
+----------+


In [81]:
%load_ext sql

# Connect to the SQLite database
%sql sqlite:///FinalDB.db
%sql SELECT COUNT(*) FROM CHICAGO_CRIME_DATA;

The sql extension is already loaded. To reload it, use:
  %reload_ext sql
   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[(533,)]

### Problem 2

##### List community area names and numbers with per capita income less than 11000.


In [93]:
%load_ext sql

# Connect to the SQLite database
%sql sqlite:///FinalDB.db

#%sql  SELECT Community_AREA FROM CENSUS_DATA;
%sql SELECT Community_AREA_NAME, COMMUNITY_AREA_NUMBER FROM CENSUS_DATA WHERE PER_Capita_Income < 11000;

The sql extension is already loaded. To reload it, use:
  %reload_ext sql
   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[('West Garfield Park', 26.0),
 ('South Lawndale', 30.0),
 ('Fuller Park', 37.0),
 ('Riverdale', 54.0)]

In [56]:
import sqlite3

# Connect to the SQLite database
conn = sqlite3.connect('FinalDB.db')

# Execute the SQL query
query = """
SELECT Community_AREA_NAME,COMMUNITY_AREA_NUMBER
FROM CENSUS_DATA
WHERE Per_Capita_Income < 11000;
"""
cursor = conn.cursor()
cursor.execute(query)

# Fetch and print the result
result = cursor.fetchall()
for row in result:
    print(row)

# Close the connection
conn.close()


('West Garfield Park', 26.0)
('South Lawndale', 30.0)
('Fuller Park', 37.0)
('Riverdale', 54.0)


### Problem 3

##### List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis) 


In [72]:
import sqlite3

# Connect to the SQLite database
conn = sqlite3.connect('FinalDB.db')  # Replace with your actual database name

# Create a cursor
cursor = conn.cursor()

# SQL query to list case numbers for crimes involving minors
query = """
SELECT CASE_NUMBER
FROM CHICAGO_CRIME_DATA
WHERE DESCRIPTION LIKE ?;
"""

# Execute the query
description_keyword = '%MINOR%'
cursor.execute(query, (description_keyword,))

# Fetch and print the results
result = cursor.fetchall()
for row in result:
    print(row[0])

# Close the connection
conn.close()



HL266884
HK238408


In [97]:
%sql SELECT CASE_NUMBER FROM CHICAGO_CRIME_DATA WHERE DESCRIPTION LIKE '%MINOR%'


   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[('HL266884',), ('HK238408',)]

### Problem 4

##### List all kidnapping crimes involving a child?


In [76]:
import sqlite3

# Connect to the SQLite database
conn = sqlite3.connect('FinalDB.db')  # Replace with your actual database name

# Create a cursor
cursor = conn.cursor()

# SQL query to list all kidnapping crimes involving a child
query = """
SELECT CASE_NUMBER, PRIMARY_TYPE, DESCRIPTION
FROM CHICAGO_CRIME_DATA
WHERE PRIMARY_TYPE = 'KIDNAPPING' AND DESCRIPTION LIKE '%CHILD%';
"""

# Execute the query
cursor.execute(query)

# Fetch and print the results
result = cursor.fetchall()
for row in result:
    print(row)

# Close the connection
conn.close()


('HN144152', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER')


In [99]:



%sql SELECT CASE_NUMBER, PRIMARY_TYPE, DESCRIPTION FROM CHICAGO_CRIME_DATA WHERE PRIMARY_TYPE = 'KIDNAPPING' AND DESCRIPTION LIKE '%CHILD%';




   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[('HN144152', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER')]

### Problem 5

##### List the kind of crimes that were recorded at schools. (No repetitions)


In [100]:
%sql SELECT DISTINCT PRIMARY_TYPE FROM CHICAGO_CRIME_DATA WHERE LOCATION_DESCRIPTION LIKE '%SCHOOL%';




   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[('BATTERY',),
 ('CRIMINAL DAMAGE',),
 ('NARCOTICS',),
 ('ASSAULT',),
 ('CRIMINAL TRESPASS',),
 ('PUBLIC PEACE VIOLATION',)]

### Problem 6

##### List the type of schools along with the average safety score for each type.


In [101]:
%sql SELECT "Elementary, Middle, or High School" as School_Type, AVG(SAFETY_SCORE) as Average_Safety_Score FROM CHICAGO_PUBLIC_SCHOOLS GROUP BY School_Type;



   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [102]:
%sql SELECT Community_Area_Name, PERCENT_HOUSEHOLDS_BELOW_POVERTY FROM CENSUS_DATA ORDER BY PERCENT_HOUSEHOLDS_BELOW_POVERTY DESC LIMIT 5;



   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[('Riverdale', 56.5),
 ('Fuller Park', 51.2),
 ('Englewood', 46.6),
 ('North Lawndale', 43.1),
 ('East Garfield Park', 42.4)]

### Problem 8

##### Which community area is most crime prone? Display the coumminty area number only.


In [103]:
%sql SELECT COMMUNITY_AREA_NUMBER FROM CHICAGO_CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY COUNT(*) DESC LIMIT 1;



   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[(25.0,)]


Double-click **here** for a hint

<!--
Query for the 'community area number' that has most number of incidents
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [104]:
%sql SELECT COMMUNITY_AREA_NAME FROM CENSUS_DATA WHERE HARDSHIP_INDEX = (SELECT MAX(HARDSHIP_INDEX) FROM CENSUS_DATA);



   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[('Riverdale',)]

In [112]:
%load_ext sql

# Connect to the SQLite database
%sql sqlite:///FinalDB.db
%sql SELECT   *FROM CHICAGO_CRIME_DATA LIMIT 2;

The sql extension is already loaded. To reload it, use:
  %reload_ext sql
   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[(3512276, 'HK587712', '2004-08-28', '047XX S KEDZIE AVE', '890', 'THEFT', 'FROM BUILDING', 'SMALL RETAIL STORE', 0, 0, 911, 9, 14.0, 58.0, '6', 1155838.0, 1873050.0, 2004, 41.8074405, -87.70395585, '(41.8074405, -87.703955849)'),
 (3406613, 'HK456306', '2004-06-26', '009XX N CENTRAL PARK AVE', '820', 'THEFT', '$500 AND UNDER', 'OTHER', 0, 0, 1112, 11, 27.0, 23.0, '6', 1152206.0, 1906127.0, 2004, 41.89827996, -87.71640551, '(41.898279962, -87.716405505)')]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [113]:
%sql SELECT BLOCK FROM CHICAGO_CRIME_DATA WHERE COMMUNITY_AREA_NUMBER = (SELECT COMMUNITY_AREA_NUMBER FROM CHICAGO_CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY COUNT(*) DESC LIMIT 1);




   sqlite:///FinalDB
 * sqlite:///FinalDB.db
Done.


[('055XX W GLADYS AVE',),
 ('058XX W ARTHINGTON ST',),
 ('017XX N AUSTIN AVE',),
 ('0000X N LATROBE AVE',),
 ('017XX N LUNA AVE',),
 ('008XX N CICERO AVE',),
 ('048XX W NORTH AVE',),
 ('017XX N NATCHEZ AVE',),
 ('056XX W ROOSEVELT RD',),
 ('051XX W MADISON ST',),
 ('047XX W SUPERIOR ST',),
 ('014XX N MENARD AVE',),
 ('048XX W HURON ST',),
 ('050XX W DIVISION ST',),
 ('005XX N WALLER AVE',),
 ('010XX N WALLER AVE',),
 ('052XX W CONGRESS PKWY',),
 ('051XX W KINZIE ST',),
 ('053XX W CONGRESS PKWY',),
 ('005XX N LAVERGNE AVE',),
 ('004XX N LARAMIE AVE',),
 ('047XX W FULTON ST',),
 ('004XX N PINE AVE',),
 ('051XX W MADISON ST',),
 ('009XX N LOREL AVE',),
 ('049XX W MAYPOLE AVE',),
 ('012XX N LOCKWOOD AVE',),
 ('016XX N MENARD AVE',),
 ('0000X N WALLER AVE',),
 ('011XX N LOREL AVE',),
 ('050XX W NORTH AVE',),
 ('050XX W JACKSON BLVD',),
 ('049XX W WASHINGTON BLVD',),
 ('017XX N MASON AVE',),
 ('068XX W NORTH AVE',),
 ('051XX W WASHINGTON BLVD',),
 ('054XX W POTOMAC AVE',),
 ('0000X S MAYFIELD AVE',),
 ('009XX N MASSASOIT AVE',),
 ('0000X N KENTON AVE',),
 ('002XX N LARAMIE AVE',),
 ('050XX W VAN BUREN ST',),
 ('059XX W CHICAGO AVE',)]

## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>
<h4>Abhishek Gagneja</h4>

## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
|2023-10-18  | 2.6     | Abhishek Gagneja  | Modified instruction set |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |

## <h3 align="center"> © IBM Corporation 2023. All rights reserved. <h3/>
